src/cryptonote_core/loki_name_system.cpp

```diff
@@ -671,11 +779,21 @@ bool name_system_db::validate_lns_tx(uint8_t hf_version, uint64_t blockchain_hei
   if (!entry) entry = &entry_;
   std::stringstream err_stream;
 
+  if (tx.type != cryptonote::txtype::loki_name_system)
+  {
+    if (reason)
+    {
+      print_tx(err_stream, tx) << ", uses wrong tx type, expected=" << static_cast<int>(cryptonote::txtype::loki_name_system);
```

Comment:
`txtype` is already stream printable, and the string value you get (e.g. "key_image_unlock") seems more helpful here than a numeric value.

Reply:
Yep. Removed cast